res/text.f.glsl

```diff
@@ -31,8 +32,14 @@ void main()
         alphaMask = vec4(1.0);
         color = vec4(bg.rgb, 1.0);
     } else {
-        vec3 textColor = texture(mask, TexCoords).rgb;
-        alphaMask = vec4(textColor, textColor.r);
-        color = vec4(fg, 1.0);
+        if (preColored != 0) {
+            vec4 c = texture(mask, TexCoords);
+            alphaMask = vec4(c.a);
+            color = c;
```

Comment:
You should apply the fix I've suggested for the second commit here, and also, don't use one char name here.